django_mesh/models.py

```diff
@@ -28,37 +29,62 @@
 
 # 3d party imports
 from model_utils import Choices
+import markdown
+import textile # to do: add oembed for textile markup
+from pyembed.markdown import PyEmbedMarkdown
+from pyembed.core import PyEmbed
+from bs4 import BeautifulSoup
 
 # App imports
 from .managers import PostQuerySet, ChannelQuerySet, TagQuerySet
 
-oembed_regex = re.compile(r'^(?P<spacing>\s*)(?P<url>http://.+)', re.MULTILINE)
-
 @python_2_unicode_compatible
 class _Abstract(models.Model):     #microblog compatible.
     slug = models.SlugField(unique=True)
     title = models.CharField(max_length=140, unique=True)
     text = models.TextField(default='')
     rendered_text = models.TextField(default='', blank=True)
+    CONVERSION = Choices(
+        (0, 'PLAINTEXT', 'Plaintext',),
+        (1, 'MARKDOWN', 'Markdown',),
+        (2, 'TEXTILE', 'Textile'),
+        )
+
+    conversion      = models.IntegerField(max_length=1, default=CONVERSION.PLAINTEXT, choices=CONVERSION)
+
+    def render(self, *args, **kwargs):
+
+        if self.conversion == self.CONVERSION.PLAINTEXT:
+
+            self.rendered_text = cgi.escape(self.text, True)
+
+        elif self.conversion == self.CONVERSION.MARKDOWN:
+
+            soup = BeautifulSoup(markdown.markdown(self.text))
```

Comment:
We'll want to set self.rendered_text here like we do for the other types, then move this BeautifulSoup code below the if/else blocks so it applies to all content types.